projects/core/src/config/config-initializer/config-initializer.module.ts

```diff
@@ -38,6 +45,13 @@ export class ConfigInitializerModule {
             [new Optional(), CONFIG_INITIALIZER],
           ],
         },
+        {
+          // Used to indicate when a location is initialized.
+          // specifically to hold on navigation before configuration is stable
```

Comment:
What about not telling what it does (because it's written as code), but adding the context for using this specialized token? Proposition:

By default, Angular Router initial navigation is scheduled AFTER the app is bootstraped (when `APP_INITIALIZER`s have been resolved; in particular: when Spartacus config is stable ).

But when Angular Router is configured differently (`initialNavigation: 'enabled'`), the initial navigation is triggered in one of the `APP_INITIALIZER`s - BEFORE the app bootstraped.

Therefore to provide more predicablitly, we provide token `LOCATION_INITIALIZED` to explicitly hold the router's initial navigation (i.e. router events, resolving guards, etc.) at least UNTIL the Spartacus config is stable.

Reply:
Thanks!